api/src/test/java/com/example/api/CountriesApiTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.api
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.toInput
+import com.apollographql.apollo.rx2.rxQuery
+import com.example.GetCountriesQuery
+import com.example.GetCountryByIdQuery
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Answers
+import org.mockito.ArgumentCaptor
+import org.mockito.Mockito.*
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class CountriesApiTest {
+
+    @Rule
+    @JvmField
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    private val apolloClientMock = mock(ApolloClient::class.java, Answers.RETURNS_DEEP_STUBS)
+
+    @Test
+    fun test_apolloClientGetCountries() {
+        val countriesApi = spy(CountriesApi(apolloClientMock))
```

Comment:
Usually, we shouldn't use `spy` for classes under tests. What was the motivation of doing this?

Reply:
Changed.